Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -1469,13 +1484,13 @@ void TextEditor::for_each_visual_line(size_t line_index, Callback callback) cons
     }
 }
 
-void TextEditor::set_line_wrapping_enabled(bool enabled)
+void TextEditor::set_wrapping_mode(WrappingMode mode)
 {
-    if (m_line_wrapping_enabled == enabled)
+    if (m_wrapping_mode == mode)
         return;
 
-    m_line_wrapping_enabled = enabled;
-    horizontal_scrollbar().set_visible(!m_line_wrapping_enabled);
+    m_wrapping_mode = mode;
+    horizontal_scrollbar().set_visible(!m_wrapping_mode);
```

Comment:
I think using either `horizontal_scrollbar().set_visible(m_wrapping_mode == WrappingMode::NoWrap)` or `horizontal_scrollbar().set_visible(!is_wrapping_enabled())` would be clearer for this ? The code works because `NoWrap == 0`, but is very unexpected.

Reply:
Good catch, that wasn't intentional, it was leftover from the previous design of the commit.